### PR TITLE
Homework5: Renat Levotskiy

### PR DESCRIPTION
**Pull request by renat33**

### CODE REVIEW

File: hw5_zadacha1.py [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+runner(‘func_name’) – вызывается только функцию func_name.
+runner(‘func’, ‘func1’...) - вызывает все переданные функции
+"""
+import hw5_func as hw5
+
+
+def runner(*call_func):
+    for el in call_func:
+        a = getattr(hw5, el)
+        a()
```
Comment (belovmd) [inline hw5_zadacha1.py:14]: what if it's not callable?

File: hw5_Zadacha2.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+import os
+
+
+def dec(a):
+    def wrapper():
+        if os.path.exists("D:/rez.txt"):
+            f = open('D:/rez.txt', "r")
+            count = int(f.read())
+            count += 1
+            print(count)
```
Comment (belovmd) [inline hw5_Zadacha2.py]: redundant print

File: hw5_Zadacha2.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+            f.close()
+            save = open('D:/rez.txt', "w")
```
Comment (belovmd) [inline hw5_Zadacha2.py:15]: you can just append
you can use context manager

File: hw5_zadacha3.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+неповторяющихся целых чисел, отсортированных по возрастанию,
+которая этот список “сворачивает”
+"""
+
+
+def get_ranges(a):
+    raw_ranges = []
+    ranges = []
+    for _ in range(len(a)):
+        for i in range(len(a)):
```
Comment (belovmd) [inline hw5_zadacha3.py:12]: O(n**2) solution. it can be solved with O(n)

File: hw5_zadacha6.py [hunk 5]
```diff
@@ -0,0 +1,15 @@
+"""
+Вводится число найти его максимальный делитель,
+являющийся степенью двойки. 10(2) 16(16), 12(4).
+"""
+user = int(input("Введите число, для определения максимального делителя "))
+
+
+def get_maxdiv(num):
```
Comment (belovmd) [inline hw5_zadacha6.py:8]: it doesn't solve task - for 10 it should be 2, not 8

**Review by belovmd — CHANGES_REQUESTED** [review]
- please fix CI checks
- please add imdb tasks results (list and histogram)
- please address next issues

**Review by belovmd — COMMENTED** [review]
please address previous review issues

File: hw5_Zadacha2.py (hunk 2 above)
Comment (renat33) [reply hw5_Zadacha2.py]: fixed

File: hw5_zadacha6.py (hunk 5 above)
Comment (renat33) [reply hw5_zadacha6.py:8]: fixed

File: hw5_zadacha1.py (hunk 1 above)
Comment (renat33) [reply hw5_zadacha1.py:14]: fixed?

File: hw5_Zadacha2.py (hunk 3 above)
Comment (renat33) [reply hw5_Zadacha2.py:15]: Didn't want to use context manager at that time.

Comment (belovmd) [reply hw5_Zadacha2.py:15]: why?

Comment (belovmd) [reply hw5_Zadacha2.py:15]: without append it doesn't keep results  for all runs (please take a look at task description

File: hw5_func.py [hunk 6]
```diff
@@ -0,0 +1,45 @@
+def get_me_languages():
```
Comment (belovmd) [inline hw5_func.py:1]: missed tasks descriptions

File: hw5_func.py [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+def zadacha7():
+    num1 = int(input())
+    num2 = int(input())
+    while num1 != 0 and num2 != 0:
+        if num1 > num2:
+            num1 = num1 % num2
+        else:
+            num2 = num2 % num1
+
+    return (num1 + num2)
```
Comment (belovmd) [inline hw5_func.py:45]: algorithm suppose one final number, not sum

File: hw5_zadacha1.py [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
```
Comment (belovmd) [inline hw5_zadacha1.py:14]: it still will fail if module will have constant and I'll pass it as arg

File: hw5_zadacha3.py (hunk 4 above)
Comment (belovmd) [reply hw5_zadacha3.py:12]: still O(n**2) solution

**Review by belovmd — COMMENTED** [review]
7

**Pull request closed without merging**